runtime/doc/lua.txt

```diff
@@ -3349,4 +3349,133 @@ totable({f}, {...})                                       *vim.iter.totable()*
     Return: ~
         (table)
 
+
+==============================================================================
+Lua module: net                                                      *lua-net*
+
+download({url}, {path}, {opts})                           *vim.net.download()*
+    Asynchronously download a file. To read the response metadata, such as
+    headers and body, use |vim.net.fetch()|.
+
+    Shares a few options with |vim.net.fetch()|, but not all of them.
+
+    Example: >lua
+
+     vim.net.download("https://.../path/file", "~/.cache/download/location", {
+       on_complete = function ()
+         vim.notify("File Downloaded", vim.log.levels.INFO)
+       end
+     })
+<
+
+    Parameters: ~
+      • {url}   (string) url
+      • {path}  (string) A download path, can be relative.
+      • {opts}  (table|nil) Optional keyword arguments:
+                • method string|nil HTTP method to use. Defaults to GET.
+                • headers table<string, string>|nil A table of key-value
+                  headers.
+                • follow_redirects boolean|nil Will follow redirects by
```

Comment:
Can we name the options match https://fetch.spec.whatwg.org/#request-class?

- `follow_redirects -> redirect="follow"`
- `data -> body`

Reply:
> - `follow_redirects -> redirect="follow"`

When naming this, I specifically chose to make it a boolean.
Until there is a way to error on redirect the other value simply has no meaning.

> - `data -> body`

Yeah, probably.

Reply:
> Until there is a way to error on redirect the other value simply has no meaning.

The only supported value can be `"follow"` right now. If the other values are implemented in future, it would be good to not rename things.